lib/ansible/modules/monitoring/grafana_dashboard.py

```diff
@@ -125,26 +132,64 @@ class GrafanaExportException(Exception):
     pass
 
 
+class GrafanaDeleteException(Exception):
+    pass
+
+
 def grafana_switch_organisation(module, grafana_url, org_id, headers):
     r, info = fetch_url(module, '%s/api/user/using/%s' % (grafana_url, org_id), headers=headers, method='POST')
     if info['status'] != 200:
         raise GrafanaAPIException('Unable to switch to organization %s : %s' % (org_id, info))
 
 
-def grafana_dashboard_exists(module, grafana_url, slug, headers):
+def grafana_headers(module, data):
+    headers = {'content-type': 'application/json; charset=utf8'}
+    if 'grafana_api_key' in data and data['grafana_api_key']:
+        headers['Authorization'] = "Bearer %s" % data['grafana_api_key']
+    else:
+        module.params['url_username'] = data.get('grafana_user')
```

Comment:
You should check the url_argument_spec function in the urls helpers.
It return an argument spec that contains all the arguments that fetch_url supports (including url_username etc ...)
You can still override the values to add grafana_user and the previous vars as alias.

Reply:
Hi, Do you mean I should use the url_argument_spec in the main function ? then update it with all the grafana params ? Ho do I create an alias for url_username <-> grafana_user ?

Reply:
> Do you mean I should use the url_argument_spec in the main function ? then update tha argument_spec var with all the grafana params ?
yes, I saw that in other modules like `web_infrastructure/jenkins_plugin.py` (line 715)

>  Ho do I create an alias for url_username <-> grafana_user ?
In the argument_spec, in the dict you have the `aliases`key. 
Check the file module `files/file.py` line 181.
The `path` parameter has aliases `dest` and `name` for backward compatibility.